graylog2-server/src/main/java/org/graylog2/inputs/InputDiagnosticService.java

```diff
@@ -0,0 +1,142 @@
+            Input input, SearchUser searchUser) {
+        final Search search = buildSearch(input);
+        final SearchJob searchJob = searchExecutor.executeSync(search, searchUser, ExecutionState.empty());
+        final SearchJobDTO searchJobDTO = SearchJobDTO.fromSearchJob(searchJob);
+        final QueryResult queryResult = searchJobDTO.results().get(QUERY_ID);
+
+        final Set<SearchError> errors = queryResult.errors();
+        if (errors != null && !errors.isEmpty()) {
+            LOG.error("An error occurred while executing aggregation: {}",
+                    errors.stream().map(SearchError::description).collect(Collectors.joining(", ")));
+            return EMPTY_DIAGNOSTICS;
+        }
+
+        final SearchType.Result aggregationResult = queryResult.searchTypes().get(PIVOT_ID);
+        if (aggregationResult instanceof PivotResult pivotResult && pivotResult.total() > 0) {
+            final List<AbstractMap.SimpleEntry<String, Long>> resultList = pivotResult.rows().stream()
+                    .filter(row -> row.source().equals("leaf"))
+                    .map(InputDiagnosticService::extractValues)
+                    .toList();
+            Map<String, Long> resultMap = new HashMap<>();
+            resultList.forEach(
+                    entry -> {
+                        try {
+                            final org.graylog2.plugin.streams.Stream stream = streamService.load(entry.getKey());
+                            resultMap.put(stream.getTitle(), entry.getValue());
+                        } catch (NotFoundException e) {
+                            LOG.warn("Unable to load stream {}", entry.getKey(), e);
+                        }
+                    }
+            );
+            return new InputDiagnostics(resultMap);
+        }
+
+        return EMPTY_DIAGNOSTICS;
+    }
+
+    private Search buildSearch(Input input) {
+        final SearchType searchType = Pivot.builder()
+                .id(PIVOT_ID)
+                .rollup(true)
+                .rowGroups(Values.builder().fields(List.of("streams")).build())
+                .series(Count.builder().build())
+                .build();
+        return Search.builder()
+                .queries(ImmutableSet.of(
+                        Query.builder()
+                                .id(QUERY_ID)
+                                .query(ElasticsearchQueryString.of(FIELD_GL2_SOURCE_INPUT + ":" + input.getId()))
+                                .searchTypes(Collections.singleton(searchType))
+                                .timerange(RelativeRange.create(900))
+                                .build()
+                ))
+                .build();
+    }
+
+    private static AbstractMap.SimpleEntry<String, Long> extractValues(PivotResult.Row r) {
+        if (r.values().size() != 1) {
+            LOG.warn("Expected 1 value in aggregation result, but received [{}].", r.values().size());
+            return null;
```

Comment:
Here, too, it might make more sense to throw an exception instead of displaying incomplete data.